### DISCUSSION TITLE
apisix kafka-logger plugin performance question

### DISCUSSION THREAD

**Discussion by windyrjc** [apache/apisix#5352] [General]
Hi, i found apisix kafka-logger plugin are using lrucache to improve performance.It using lrucache to create producer instance and default plugin lrucache count is 8, but in lua-resty-kafka module, every producer instance are using ngx.timer.at every 5 second to sync kafka metadata info.  Is this unnecessary? Or can we have a better way to handle this client instance data sync in lrucache situation?

**Comment by tokers** [comment]
Didn't capture your question, is the number of producers too large? Or is the timer period too narrow?
